lib/expand.js

```diff
@@ -346,7 +346,7 @@ class Expander {
       if (targetVal instanceof models.ChoiceValue) {
         let isValidOption = false;
         for (const opt of targetVal.aggregateOptions) {
-          if (opt instanceof models.IdentifiableValue && this.checkHasBaseType(constraint.isA, opt.identifier)) {
+          if (opt instanceof models.IdentifiableValue && (opt.identifier.equals(constraint.isA) || this.checkHasBaseType(constraint.isA, opt.identifier))) {
```

Comment:
This clearly works for this case, but I wonder if it would be better to put this logic directly in `checkHasBaseType` -- that way other bits of code that use `checkHasBaseType` also benefit.

I looked at the implementation of `checkHasBaseType` and it uses `getRecursiveBaseTypes`.  For non-primitives, the `getRecursiveBaseTypes` returns the root identifier along with its base types -- but for primitives, there is some short-circuiting logic which causes the primitive _not_ to be returned in the result.  I think this was probably a mistake -- and fixing it would also cause the desired behavior here as well (allowing you to remove the extra equality check here).

What do you think?

Reply:
So to be clear, you actually wouldn't have to modify this line, or `checkHasBaseType`, but rather, you'd have to modifier `getRecursiveBaseTypes` to return primitives as well (when it short-circuits to avoid infinite recursion).